### PR TITLE
kubernetes modify statefulset controller, make code clear.

### PR DESCRIPTION
**Pull request by jacky68147527**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
kubernetes modify statefulset controller, make code clear, simplified code judgment logic.

```release-note
NONE
```

### CODE REVIEW

**Comment by jacky68147527** [comment]
I signed it

**Comment by Joseph-Irving** [comment]
/kind cleanup
/priority important-soon
/lgtm

Please add release none, see https://git.k8s.io/community/contributors/guide/release-notes.md

**Comment by jacky68147527** [comment]
@Joseph-Irving thank you, i had added the release note. Could you help me to merge the pr ?

**Comment by Joseph-Irving** [comment]
You'll need somebody with approval powers to get this merged 
/assign @tnozicka
cc @kubernetes/sig-apps-pr-reviews

**Comment by jacky68147527** [comment]
cc /assign @smarterclayton @tnozicka @soltysh Could you help me to merge the pr ?

File: pkg/controller/statefulset/stateful_set_control.go
Comment (tnozicka) [inline pkg/controller/statefulset/stateful_set_control.go:229]: The code avoided nesting. How is this PR helping our code base other then presenting different opinion on the structure then the original author?

**Review by tnozicka — CHANGES_REQUESTED** [review]

File: pkg/controller/statefulset/stateful_set_control.go
Comment (jacky68147527) [reply pkg/controller/statefulset/stateful_set_control.go:229]: thanks @tnozicka 
Good question !
This PR is to change the struct like

```
if A && B {
	
} else if A {
	
} else {
	
}
```

to the struct like

```
if A {
	if B {

	} else {

	}
} else {
	
}
```

From Boolean logic, we can be sure that there will be no moral change. 
1. Such nesting, we can easy to see that the outer `if... else...` is to judge whether `A ` is true, and the inner is to judge whether `B` is true on this basis. In the original way, it is not easy to judge the exact logical meaning represented by else if `A`.
2. When `A` is false, the original writing method needs to judge if and else if, because them both contain `A` , while PR's writing method does not, it only needs to be judged once.

So I think this PR is more neat and understandable.

Comment (soltysh) [reply pkg/controller/statefulset/stateful_set_control.go:229]: I agree with Tomas here, although from logical point of view nothing changes, the readability actually hurts. The preferred approach is to have less nesting than needed. Your code goes the other direction. I could be convinced in favor of this change, but only if the calculation of `A` from your example was computationally expensive, here it's just a simple comparison, which is negligible. So I don't think this provides any value.

**Review by soltysh — CHANGES_REQUESTED** [review]
/lgtm cancel
/hold

File: pkg/controller/statefulset/stateful_set_control.go
Comment (jacky68147527) [reply pkg/controller/statefulset/stateful_set_control.go:229]: Thanks @soltysh 
I understand what you mean, i close this PR now, thanks again!

**Pull request closed without merging**
